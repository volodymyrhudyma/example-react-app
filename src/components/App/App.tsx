import React, { Component } from 'react';
import { GlobalStyle } from 'src/styled';
import { Wrapper, ContentWrapper } from './styled/App.styled';
import Navbar from 'src/ui-kit/Navbar';
import { menuItems } from 'src/constants/navbar';
import Header from 'src/components/Header';
import Content from 'src/components/Content';

class App extends Component {
  render() {
    return (
      <Wrapper>
        <GlobalStyle />
        <Navbar items={menuItems} />
        <ContentWrapper>
          <Header />
          <Content />
        </ContentWrapper>
      </Wrapper>
    );
  }
}

export default App;
