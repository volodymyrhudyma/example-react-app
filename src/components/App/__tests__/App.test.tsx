import { shallow } from 'enzyme';
import React from 'react';
import App from '../App';

describe('App with Enzyme and Jest', () => {
  it('should render correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
