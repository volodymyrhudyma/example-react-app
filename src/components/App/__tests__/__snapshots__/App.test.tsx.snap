// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App with Enzyme and Jest should render correctly 1`] = `
<styled.div>
  <GlobalStyleComponent />
  <Navbar
    items={
      Array [
        Object {
          "icon": "/icons/dashboard.svg",
          "link": "#",
        },
        Object {
          "icon": "/icons/sessions.svg",
          "link": "#",
        },
        Object {
          "icon": "/icons/meetings.svg",
          "link": "#",
        },
        Object {
          "icon": "/icons/files.svg",
          "link": "#",
        },
        Object {
          "icon": "/icons/conversations.svg",
          "link": "#",
        },
        Object {
          "icon": "/icons/settings.svg",
          "link": "#",
        },
      ]
    }
  />
  <styled.div>
    <withI18nextTranslation(Header) />
    <Connect(withI18nextTranslation(Content)) />
  </styled.div>
</styled.div>
`;
